nifi-bootstrap/src/main/java/org/apache/nifi/bootstrap/RunNiFi.java

```diff
@@ -324,27 +324,46 @@ private void registerNotificationServices(final NotificationServiceManager manag
         defaultLogger.info("Registered {} Notification Services for Notification Type {}", registered, type);
     }
 
-    File getStatusFile() {
+    File getStatusFile() throws IOException{
         return getStatusFile(defaultLogger);
     }
 
-    public File getStatusFile(final Logger logger) {
+    public File getStatusFile(final Logger logger) throws IOException{
         final File confDir = bootstrapConfigFile.getParentFile();
         final File nifiHome = confDir.getParentFile();
-        final File bin = new File(nifiHome, "bin");
-        final File statusFile = new File(bin, "nifi.pid");
 
-        logger.debug("Status File: {}", statusFile);
+        String pidDir = System.getProperty(NIFI_PID_DIR_PROP);
+
+        final File pidFileDir;
 
+        if(pidDir != null){
+            pidFileDir = new File(pidDir.trim());
+        } else{
+            pidFileDir = new File(nifiHome, DEFAULT_PID_DIR);
+        }
+
+        FileUtils.ensureDirectoryExistAndCanAccess(pidFileDir);
+        final File statusFile = new File(pidFileDir, "nifi.pid");
+        logger.debug("Status File: {}", statusFile);
         return statusFile;
     }
 
-    public File getLockFile(final Logger logger) {
+    public File getLockFile(final Logger logger) throws IOException{
```

Comment:
This method's logic is identical to the one above with the exception of the file extension. To reduce code duplication, perhaps we could provide a private method with the logic that just accepts the filename, and both could invoke that with a hard-coded value (even better, constants).